### ISSUE TITLE
routeRules not working

### ISSUE THREAD

**Issue by tmlmt** [harlan-zw/nuxt-simple-robots#6]
### Describe the bug

`routeRules` (neither `{ index: ... }` nor `{ robots: ... } `) are not taken into account. Only the `robots` config with `disallow` leads to a `robots.txt` file different than: 
```
# START nuxt-simple-robots (indexable: true)
# This is a development preview
User-agent: *
Disallow: 

# END nuxt-simple-robots
```
Is this the expected behavior when using routeRules? Looks like it when looking at [source code for robots.txt](https://github.com/harlan-zw/nuxt-simple-robots/blob/a3977b254b34f99c3bfb393993dade2f74e4582a/src/runtime/server/robots-txt.ts). The doc currently says that injecting meta tags is optional:
> This shouldn't be necessary, but it's useful if you want to be extra sure that your page is not indexed and can make debugging easier.

### Reproduction

https://stackblitz.com/edit/nuxt-starter-j2xojn

### System / Nuxt Info

```Shell
------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.3.1`
- Nitro Version:    `2.3.1`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `modules`, `robots`, `routeRules`
- Runtime Modules:  `nuxt-simple-robots@2.0.1`
- Build Modules:    `-`
------------------------------
```

**Comment by harlan-zw** [comment]
Hey @tmlmt, thanks for the issue.

The logic is like this to avoid leaking page paths which may be intentionally hidden for security reasons. For example, if we have an admin login on our site, it's probably a good idea not to tell people where to find it. (Though security through obscurity is not a good measure).

It's not a big problem, but personally, this is how I like it handled.

When you use `index: false`, it will add the Robots header through the middleware. The only issue with that is it's not going to work for SSG apps, in which case you need the meta tags, but this should be documented better.

If you really need something in robots.txt you can use the `disallow` config key it will add it to robots.txt, so you still have some control.

Would be great to hear your thoughts on this as I'm undecided on the best approach.

**Comment by harlan-zw** [comment]
FYI docs are updated to make this clearer

**Comment by tmlmt** [comment]
Hey @harlan-zw, my understanding is that there are multiple ways to instruct crawlers, (i) robots.txt, (ii) HTTP Robots header and (iii) HTML meta tags, which are as many options that they can disregard, so there's no bulletproof way to present page leaking. In which case it might be down to personal choice, and I personally like the simple way to place everything into robots.txt

Bottomline, the important thing is that the behavior of nuxt-simple-robots's config is clear in the docs, and that's what you just did I think. So I'd be ready to close this :)

**Comment by tmlmt** [comment]
Note: did you mean `disallow` instead of `exclude`?

**Comment by harlan-zw** [comment]
> Note: did you mean `disallow` instead of `exclude`?

I did thanks!

I've pushed out v2.1.0 which includes an option `disallowNonIndexableRoutes`. When enabled it will automatically add non-indexed route rules to the sitemap. This also included some improvements to the documentation around route rules.

```ts
export default defineNuxtConfig({
  robots: {
    disallowNonIndexableRoutes: true
  },
  routeRules: {
    '/about': { index: false },
  },
})
```

```txt
User-agent: *
Disallow: /about
```


This is disabled by default for now but I may consider enabling it by default in a v3.

Let me know if you have any feedback on it :) Thanks again for the issue.
